### ISSUE TITLE
freebayes 0.9.21-19: unable to find FASTA index entry for chrM

### ISSUE THREAD

**Issue by meiers** [freebayes/freebayes#185]
Hi there,
I am having the same problem as #159, but I have a clue what the reason could be:

I aligned my reads to the "full" reference, including various contigs and chrM. 
Later, when I call variants with freebayes, I specify a reduced reference to the `-f` flag, containing only a few chromosomes, because I am only interested in the SNPs on those chromosomes. My gut feeling tells me that as soon as freebayes reaches the first chr in the bam file which does not occur in the reference (that is chrM in my case) the error pops up. 
If that's the reason it should be easy to fix (if you know the code), right?
Hope this is helpful.

Cheers,
Sascha

**Comment by ekg** [comment]
Try running one job per chromosome that you are interested in. Then combine
the results.

**Comment by meiers** [comment]
Thanks, that works for me.
Sascha

**Comment by luciagrami** [comment]
Hello, I have a similar problem, the error looks like:

unable to find FASTA index entry for 'NZ_CP009058'

But in this case my reference file is complete, I used bwa index for indexing the reference.fa and then bwa and samtools always usigng the complete reference. Also I would like to have the SNPs for every contig in the reference. 
I used an annotated reference file and it worked fine.
What could be the problem??

Thanks!!
